zerver/lib/upload.py

```diff
@@ -70,6 +77,8 @@
 # "file name" is the original filename provided by the user run
 # through a sanitization function.
 
+redis_client = get_redis_client()
```

Comment:
Does this create a separate Redis connection for every file that calls `get_redis_client()`?

We could use Django’s [`TimestampSigner`](https://docs.djangoproject.com/en/2.2/topics/signing/#verifying-timestamped-values) instead of using a database.

Reply:
No, this runs at import time.  But I think it'd still be better to use `TimestampSigner`; that's a cleaner way to do it than custom redis code for the same goal.

Reply:
> We could use Django’s [`TimestampSigner`](https://docs.djangoproject.com/en/2.2/topics/signing/#verifying-timestamped-values) instead of using a database.

Ah, handy!

I had the thought that an HMAC would be a good way to do this without having to store anything, and didn't realize that Django provides something so conveniently ready-made to do the job.

Reply:
I know it runs at import time, but what I’m saying is that if we have four files that call `get_redis_client()` at import time, it seems we’ll have four different Redis connections for no reason (unless there’s some automatic pooling that I don’t understand?).

Reply:
One gotcha that I learned from reading the code, which the docs don't adequately stress: when using that facility, we should always pass the `salt` parameter, with a value that's unique basically per call site of `sign` in the app source code.

(Otherwise, if in two unrelated pieces of code we happen to use it to sign values with the same shape but different meanings, then a signed blob from one can be used against the other.)

Easy to handle, so not a reason to not use it -- just need to be sure to do so.

Reply:
> when using that facility, we should always pass the `salt` parameter, with a value that's unique basically per call site of `sign` in the app source code.

I.e. we'd use something just like the `LOCAL_FILE_ACCESS_TOKEN_PREFIX` in the current version of the PR, and just pass that as the salt.

Reply:
`get_redis_client` doesn't make a connection when called, only returns an instance of `redis.Redis` class. Connections are made when necessary - so when a command gets actually called the first time on the `Redis` instance. But yes, I think we'll have separate connections for each. I've been actually wondering if we shouldn't just instantiate one `redis_client` in `redis_utils.py` and have files that need redis access just import it from there.

With that said, TimestampSigner looks more suitable here indeed so I'll change that